joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/passes/MethodAndTypeCacheBuilderPass.scala

```diff
@@ -1,30 +1,40 @@
 package io.joern.gosrc2cpg.passes
 
 import io.joern.gosrc2cpg.Config
-import io.joern.gosrc2cpg.astcreation.{AstCreator, CacheBuilder}
+import io.joern.gosrc2cpg.astcreation.AstCreator
 import io.joern.gosrc2cpg.model.GoModHelper
 import io.joern.gosrc2cpg.parser.GoAstJsonParser
-import io.joern.gosrc2cpg.parser.GoAstJsonParser.ParserResult
 import io.joern.x2cpg.SourceFiles
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.DiffGraphBuilder
 
 import java.nio.file.Paths
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.Duration
 import scala.concurrent.{Await, Future}
 
-class MethodAndTypeCacheBuilderPass(astFiles: List[String], config: Config, goMod: GoModHelper) {
+class MethodAndTypeCacheBuilderPass(cpgOpt: Option[Cpg], astFiles: List[String], config: Config, goMod: GoModHelper) {
   def process(): Seq[AstCreator] = {
     val futures = astFiles
       .map(file => {
         Future {
           val parserResult    = GoAstJsonParser.readFile(Paths.get(file))
           val relPathFileName = SourceFiles.toRelativePath(parserResult.fullPath, config.inputPath)
           val astCreator      = new AstCreator(relPathFileName, parserResult, goMod)(config.schemaValidation)
-          astCreator.buildCache()
-          astCreator
+          val diffGraph       = astCreator.buildCache()
+          (astCreator, diffGraph)
         }
       })
-    val allResults: Future[List[AstCreator]] = Future.sequence(futures)
-    Await.result(allResults, Duration.Inf)
+    val allResults: Future[List[(AstCreator, DiffGraphBuilder)]] = Future.sequence(futures)
+    val results                                                  = Await.result(allResults, Duration.Inf)
+    val (astCreators, diffGraphs)                                = results.unzip
+    cpgOpt.map(cpg => {
```

Comment:
Stylistically speaking, I prefer 
```scala
cpgOpt.map { cpg =>
...
    diffGraphs.foreach { diffGraph =>
    ...
```

Reply:
@DavidBakerEffendi Is there something that needs to be changed here?

Reply:
Yes, from a style perspective it is inconsistent with the rest of Joern code, the code is `cpgOpt.map(cpg => {` instead of `cpgOpt.map { cpg =>`

Reply:
For time's sake, I'll merge it as is for now and this can be changed in a follow-up

Reply:
Ohh I get it. I didn't notice the braces